### PR TITLE
Feature/domain asset permission

### PR DESCRIPTION
**Pull request by grimadas**
### Description of the Change

This PR adds domain permissions for commands: AddAssetQty and SubtractAssetQty.

### Benefits

Following changes allow restricting permissions for issuing asset for specific domains. For example, one can create assets only in their domain if has this permission. 

### Possible Drawbacks 

None

### CODE REVIEW

**Comment by neewy** [comment]
I would change the target into trunk, where we can update documentation as well @grimadas

File: irohad/ametsuchi/impl/postgres_command_executor.cpp [hunk 1]
```diff
@@ -68,6 +69,8 @@ namespace {
     return {with_validation.str(), without_validation.str()};
   }
 
+  auto hasCommandPermission() {}
```
Comment (Akvinikym) [inline irohad/ametsuchi/impl/postgres_command_executor.cpp]: What it is this?

Comment (grimadas) [reply irohad/ametsuchi/impl/postgres_command_executor.cpp]: Thanks, removed

File: irohad/ametsuchi/impl/postgres_command_executor.cpp [hunk 2]
```diff
@@ -264,6 +265,31 @@ namespace {
...
+      shared_model::interface::permissions::Role domain_role,
+      const shared_model::interface::types::AccountIdType &creator_id,
+      const shared_model::interface::types::AccountIdType
+          &id_with_target_domain) {
+    std::string query = (boost::format(R"(WITH
+          has_global_role_perm AS (%1%),
+          has_domain_role_perm AS (%2%)
+          SELECT CASE
+                           WHEN (SELECT * FROM has_global_role_perm) THEN true
+                           WHEN ((split_part(%3%, '@', 2) = split_part(%4%, '@', 2))
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/postgres_command_executor.cpp]: I think not this WHEN never used. You can change the type of id_with_target_domain to AssetIdType

File: test/module/irohad/ametsuchi/postgres_executor_test.cpp [hunk 3]
```diff
@@ -103,6 +103,23 @@ namespace iroha {
                         true)));
       }
 
+      void addOnePerm(
```
Comment (kamilsa) [inline test/module/irohad/ametsuchi/postgres_executor_test.cpp]: Leave a comment please

**Pull request merged**